database/access_rules.go

```diff
@@ -0,0 +1,283 @@
+// Manages Access Rules for a DBaaS Service Instance.
+// The only fields that can be Updated for an Access Rule is the desired state
+// of the access rule. From Enabled -> Disabled.
+// Deleting an Access Rule also requires an Update call, instead of a Delete API request,
+// but the Operation body parameter changes from `update` to `delete`.
+// All other parameters for the resource, aside from Status should be ForceNew.
+// The READ function for the AccessRule resource is tricky, as there is
+// no exposed `GET` function on the AccessRule API.
+// There is an API endpoint to view "all" rules, however, which will be used as a
+// data source to match on a supplied AccessRule name.
+// Timeout only supported for the CREATE method
+
+package database
+
+import "time"
+
+// API URI Paths for Container and Root objects
+const (
+	DBAccessContainerPath = "/paas/api/v1.1/instancemgmt/%s/services/dbaas/instances/%s/accessrules"
+	DBAccessRootPath      = "/paas/api/v1.1/instancemgmt/%s/services/dbaas/instances/%s/accessrules/%s"
+)
+
+// Default Timeout value for Create
+const WaitForAccessRuleTimeout = time.Duration(10 * time.Second)
+
+// AccessRules returns a UtilityClient for managing SSH Keys and Access Rules for a DBaaS Service Instance
+func (c *DatabaseClient) AccessRules() *UtilityClient {
+	return &UtilityClient{
+		UtilityResourceClient: UtilityResourceClient{
+			DatabaseClient:   c,
+			ContainerPath:    DBAccessContainerPath,
+			ResourceRootPath: DBAccessRootPath,
+		},
+	}
+}
+
+// Status Constants for an Access Rule
+type AccessRuleStatus string
+
+const (
+	AccessRuleEnabled  AccessRuleStatus = "enabled"
+	AccessRuleDisabled AccessRuleStatus = "disabled"
+)
+
+// Operational Constants for either Updating/Deleting an Access Rule
+type AccessRuleOperation string
+
+const (
+	AccessRuleUpdate AccessRuleOperation = "update"
+	AccessRuleDelete AccessRuleOperation = "delete"
+)
+
+// Default Destination for an Access Rule
+type AccessRuleDestination string
+
+const (
+	AccessRuleDefaultDestination AccessRuleDestination = "DB"
+)
+
+// Used for the GET request, as there's no direct GET request for a single Access Rule
+type AccessRules struct {
+	Rules []AccessRuleInfo `json:"accessRules"`
+}
+
+// AccessRuleInfo holds all of the known information for a single AccessRule
+type AccessRuleInfo struct {
+	// The Description of the Access Rule
+	Description string `json:"description"`
+	// The destination of the Access Rule. Should always be "DB".
+	Destination AccessRuleDestination `json:"destination"`
+	// The ports for the rule.
+	Ports string `json:"ports"`
+	// The name of the Access Rule
+	Name string `json:"ruleName"`
+	// The Type of the rule. One of: "DEFAULT", "SYSTEM", or "USER".
+	// Computed Value
+	RuleType string `json:"ruleType"`
```

Comment:
should this not be returning a Constant/Enum?

Reply:
Definitely can! Will fix